cpp/src/replace/clamp.cu

```diff
@@ -0,0 +1,206 @@
+    operator()(column_view const& input,
+              scalar const& lo,
+              scalar const& hi,
+              rmm::mr::device_memory_resource* mr,
+              cudaStream_t stream) {
+        using ScalarType = cudf::experimental::scalar_type_t<T>;
+        auto lo_scalar = static_cast<ScalarType const*>(&lo);
+        auto hi_scalar = static_cast<ScalarType const*>(&hi);
+        auto lo_value = lo_scalar->value(stream);
+        auto hi_value = hi_scalar->value(stream);
+        // mask will not change
+        std::unique_ptr<column> output {nullptr};
+        if (input.nullable()){
+            output = make_fixed_width_column(input.type(), input.size(),
+                                             copy_bitmask(input, stream, mr),
+                                             input.null_count(), stream, mr);
+        } else {
+            output = allocate_like(input, input.size(), mask_allocation_policy::NEVER, mr, stream);
+        }
+
+        auto output_device_view  = cudf::mutable_column_device_view::create(output->mutable_view(), stream);
+        auto input_device_view  = cudf::column_device_view::create(input, stream);
+
+        if (lo.is_valid(stream) and hi.is_valid(stream)) {
+            auto pred = [lo_value, hi_value] __device__ (T input, bool is_valid = true){
+              if (is_valid) {
+                  if (input < lo_value) {
+                      return lo_value;
+                  } else if (input > hi_value) {
+                      return hi_value;
+                  }
+              }
+
+              return input;
+            };
+
+            apply_transform<T>(*input_device_view, *output_device_view, pred, stream);
+
+        } else if (not lo.is_valid(stream)) {
+            auto pred = [hi_value] __device__ (T input, bool is_valid = true){
+                if (is_valid and input > hi_value) {
+                    return hi_value;
+                }
+
+                return input;
+            };
+
+            apply_transform<T>(*input_device_view, *output_device_view, pred, stream);
+
+        } else {
+
+            auto pred = [lo_value] __device__ (T input, bool is_valid = true){
+
+                if (is_valid and input < lo_value) {
+                    return lo_value;
+                }
+
+                return input;
+            };
+
+            apply_transform<T>(*input_device_view, *output_device_view, pred, stream);
+
+        }
+        
+        return output;
+    }
+    
+    template <typename T>
+    std::enable_if_t<not cudf::is_fixed_width<T>(), std::unique_ptr<cudf::column>>
+    operator()(column_view const& input,
+              scalar const& lo,
+              scalar const& hi,
+              rmm::mr::device_memory_resource* mr,
+              cudaStream_t stream) {
+        CUDF_FAIL("Not yet implemented for string");
+
+    }
+
+};
+}
+
+/**
+ * @brief Returns a column formed from `input`, where values less than
```

Comment:
Rather than repeating the public APII docs in the detail API, consider using 

```
* @copydoc cudf::clamp
```

http://www.doxygen.nl/manual/commands.html#cmdcopydoc

Reply:
But, detail will have stream parameter and if we use copydoc, that parameter would be shown after return if I am not wrong.

Reply:
You can add to the docs that are copied by adding more text.  e.g.

```
* @copydoc cudf::clamp
*
* @param stream ...
```

If you are worried about the order they will be shown in the HTML docs, don't -- doxygen groups parameters together separately from other things like returns and template params.

For example: 
https://thrust.github.io/doc/group__transformations_gaa41d309b53fa03bf13fe35a184148400.html#gaa41d309b53fa03bf13fe35a184148400